workflow-ui/core-android/src/main/java/com/squareup/workflow1/ui/navigation/DialogCollator.kt

```diff
@@ -228,26 +232,23 @@ internal class DialogCollator {
           val (id, session) = establishedSessionsIterator.next()
           if (idAndUpdates.id == id && session.canShow(overlay)) return@OldSessionFinder session
 
-          // Can't update this session from this Overlay. Dismiss it (via Dialog.dismiss
-          // under the hood), but hold on to it in case it can except a later Overlay.
-          session.setVisible(false)
-          hiddenSessions.add(Pair(id, session))
+          // Can't update this session from this Overlay, save its view state and destroy it.
+          // It it was just out of z order, a new one with matching id will be made and restored.
```

Comment:
Supernit: If it

Reply:
Fair. The good news is that I realized I need to fix these because various tests broke. I'm pretty confident.